host/storage/storage.go

```diff
@@ -967,12 +994,24 @@ func NewVolumeManager(vs VolumeStore, a Alerts, cm ChainManager, log *zap.Logger
 
 		volumes:        make(map[int]*volume),
 		changedVolumes: make(map[int]bool),
+		cache:          cache,
+		cacheMu:        sync.Mutex{},
 
 		tg: threadgroup.New(),
 	}
 	if err := vm.loadVolumes(); err != nil {
 		return nil, err
 	}
+
+	// Background goroutine to evict old entries from cache
+	go func() {
```

Comment:
?

Reply:
Yes, you are right.The background goroutine that sleeps for a minute and locks/unlocks the mutex does not actually do anything useful and can be removed.